src/api_gen.jl

```diff
@@ -43,37 +101,28 @@ end
 
 NvimApiObject(c, e::Ext) = NvimApiObject{int(e.typecode)}(c, e.data)
 #Not really module-interface clean, I know...
-function MsgPack.pack{N}(s, o::NvimApiObject{N})
-    MsgPack.pack(s, Ext(N, o.hnd))
-end
-
-
-# FIXME: the elephant in the room (i.e. handle &encoding)
-retconvert(c,val::Dict) = Dict{Any,Any}([(retconvert(c,k),retconvert(c,v)) for (k,v) in val])
-retconvert(c,val::Vector{Uint8}) = bytestring(val)
-retconvert(c,val::Vector) = [retconvert(c,v) for v in val]
-retconvert(c,val::Ext) = NvimApiObject(c, val)
-retconvert(c,val) = val
-
+MsgPack.pack{N}(s, o::NvimApiObject{N}) = MsgPack.pack(s, Ext(N, o.hnd))
+MsgPack.pack(s, t::Tuple) = MsgPack.pack(s, [v for v in t])
```

Comment:
As this is a very general rule, I think it should be rather upstreamed to `Msgpack.jl`. (Then it could be done more efficiently than needing to allocate an array, as well). I will implement this as a converter method for now.